src/daft-scan/src/scan_task_iters.rs

```diff
@@ -111,3 +116,92 @@ impl Iterator for MergeByFileSize {
         }
     }
 }
+
+pub fn split_by_row_groups(scan_tasks: BoxScanTaskIter, max_tasks: usize) -> BoxScanTaskIter {
+    let mut scan_tasks = itertools::peek_nth(scan_tasks);
+
+    // only split if we have a small amount of files
+    if scan_tasks.peek_nth(max_tasks).is_some() {
+        return Box::new(scan_tasks);
+    }
+
+    Box::new(
+        scan_tasks
+            .map(|t| -> DaftResult<BoxScanTaskIter> {
+                let t = t?;
+
+                // only split parquet tasks if they:
+                // - have one source
+                // - use native storage config
+                // - have no specified chunk spec or number of rows
```

Comment:
We also only would want to split (and fetch metadata) if the file size if over a threshold.